### DISCUSSION TITLE
Local playback with caching (offline storage)

### DISCUSSION THREAD

**Discussion by curiousercreative** [jcorporation/myMPD#1021] [Q&A] [answered]
Hello, I've just stumbled upon this project and from the documentation you seem to have most of my use cases supported, so our needs must be quite similar :) Anyhow, I've a week or two testing various combinations of music servers (mpd and mopidy) as well as music streaming servers (Navidrome, gonic, Funkwhale). Right now, I'm struggling to find the goldilocks solution that checks these boxes:
1. a single library with playlists and favorites
1. jukebox/server audio output, preferably multi-room with output controls integrated into client
1. local audio output that is cache-able proactively (offline storage on a cell phone for limited data plans or loss of connectivity)

There are a great number of other features that I am appreciating, but this particular combo is difficult. How do you handle that third desire for cache-able local audio output? My latest solution for the three desires that I'm testing now is:
1. Navidrome (or gonic, etc)
2. mopidy + mopidy-subidy (doesn't support the multi-room output controls AFAIK)
3. Navidrome + DSub/Substreamer (I have issues with both of these Android clients at the moment, but somewhat workable)

**Comment by jcorporation** [comment, marked as answer]
myMPD is only a client for MPD. There is no local playback capability without a connection to MPD.
